src/actions.js

```diff
@@ -188,6 +189,73 @@ export const removePromoCode = () => (dispatch, getState) => {
     }
 }
 
+export const validatePromoCode = (ticketData) => async (dispatch, getState, { apiBaseUrl, getAccessToken }) => {
+
+    const { registrationLiteState: { settings: { summitId }, promoCode: currentPromoCode } } = getState();    
+
+    if (summitId && currentPromoCode) {
+
+        dispatch(startWidgetLoading());
+
+        const { ticketQuantity, id, sub_type } = ticketData;
+
+        const access_token = await getAccessToken();
+
+        let params = {
+            access_token,
+            filter: `ticket_type_id==${id},ticket_type_qty==${ticketQuantity},ticket_type_subtype==${sub_type}`
+        };
+
+        const errorHandler = (err, res) => (dispatch, state) => {
+            if (res && res.statusCode === 412) {
```

Comment:
u need to manage here 404 and 429 ( rate limit [ too many attemps see https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/429])